nix/pkgs/main/default.nix

```diff
@@ -66,7 +66,11 @@ commonAttrs = {
       # right thing here.
       pkgsBuildHost.rustPlatform.bindgenHook
   ]
-  ++ lib.optionals stdenv.isDarwin [ libiconv ];
+      # https://github.com/NixOS/nixpkgs/issues/206242
+  ++ lib.optionals stdenv.isDarwin [ libiconv ]
+      # https://stackoverflow.com/questions/69869574/properly-adding-darwin-apple-sdk-to-a-nix-shell
+      # https://discourse.nixos.org/t/compile-a-rust-binary-on-macos-dbcrossbar/8612
+  ++ lib.optionals stdenv.isDarwin [ pkgsBuildHost.darwin.apple_sdk.frameworks.Security ];
```

Comment:
By the way in pseudocode this is basically
```
concat
if stdenv.isDarwin:
  [libiconv]
else:
  []
concat
if stdenv.isDarwin:
  [pkgsBuildHost.darwin.apple_sdk.frameworks.Security]
else:
  return []
```
so if you wanted to you could instead add both of them to one list like
```nix
++ lib.optionals stdenv.isDarwin [ 
    # https://github.com/NixOS/nixpkgs/issues/206242
    libiconv
    # https://stackoverflow.com/questions/69869574/properly-adding-darwin-apple-sdk-to-a-nix-shell
    # https://discourse.nixos.org/t/compile-a-rust-binary-on-macos-dbcrossbar/8612
    pkgsBuildHost.darwin.apple_sdk.frameworks.Security
];